test/resolver/bir/node_modules/bar/package.json

```diff
@@ -0,0 +1,3 @@
+{
+  "main": "index.js"
```

Comment:
In this case, the "main" being "index.js" is why it's failing. If the "main" is "index", or ".", or "./", then the extension resolution will kick in.

I've made this change and pushed it up, so you can see that the tests pass.